jbpm-designer-backend/src/main/java/org/jbpm/designer/web/server/CalledElementServlet.java

```diff
@@ -177,6 +182,108 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         }
 	}
 
+	// package scope in order to test the method
+	List<String> getRuleFlowNames(HttpServletRequest req) {
+	    final String [] projectAndBranch  = getProjectAndBranchNames(req);
+
+        // Query RuleFlowGroups for asset project and branch
+        List<RefactoringPageRow> results = queryService.query(
+                FindRuleFlowNamesQuery.NAME,
+                new HashSet<ValueIndexTerm>() {{
+                    add(new ValueSharedPartIndexTerm("*", PartType.RULEFLOW_GROUP, TermSearchType.WILDCARD));
+                    add(new ValueProjectNameIndexTerm(projectAndBranch[0]));
+                    if( projectAndBranch[1] != null ) {
+                        add(new ValueBranchNameIndexTerm(projectAndBranch[1]));
+                    }
+                }});
+
+        final List<String> ruleFlowGroupNames = new ArrayList<String>();
+        for ( RefactoringPageRow row : results ) {
+            ruleFlowGroupNames.add( (String) row.getValue() );
+        }
+        Collections.sort( ruleFlowGroupNames );
+
+        // Query RuleFlowGroups for all projects and branches
+        results = queryService.query(
+                FindRuleFlowNamesQuery.NAME,
+                new HashSet<ValueIndexTerm>() {{
+                    add(new ValueSharedPartIndexTerm("*", PartType.RULEFLOW_GROUP, TermSearchType.WILDCARD));
+                }});
+        final List<String> otherRuleFlowGroupNames = new LinkedList<String>();
+        for ( RefactoringPageRow row : results ) {
+            String ruleFlowGroupName = (String) row.getValue();
+            if( ! ruleFlowGroupNames.contains(ruleFlowGroupName) ) {
+                // but only add the new ones
+                otherRuleFlowGroupNames.add( ruleFlowGroupName );
+            }
+        }
+        Collections.sort( otherRuleFlowGroupNames );
+
+        // divider between asset project/branch ruleflowgroup names and all ruleflowgroup names
+        ruleFlowGroupNames.add("********");
```

Comment:
I can't see that the old way had any demarcation between the two sets of Rule Flow Group Names. If you're adding the distinction I'd assume something somewhere client-side would need tweaking to handle it too?